matrix/Quaternion.hpp

```diff
@@ -44,14 +58,72 @@ class Quaternion : public Vector<Type, 4>
         q(3) = 0;
     }
 
+    /**
+     * Constructor from Matrix41
+     *
+     * @param other Matrix41 to copy
+     */
     Quaternion(const Matrix41 & other) :
         Vector<Type, 4>(other)
     {
     }
 
+    /**
+     * Constructor from dcm
+     *
+     * Instance is initialized from a dcm representing coordinate transformation
+     * from frame 2 to frame 1.
+     *
+     * @param dcm dcm to set quaternion to
+     */
     Quaternion(const Dcm<Type> & dcm) :
         Vector<Type, 4>()
     {
+        set_from_dcm(dcm);
+    }
+
+    /**
+     * Constructor from euler angles
+     *
+     * This sets the instance to a quaternion representing coordinate transformation from
+     * frame 2 to frame 1 where the rotation from frame 1 to frame 2 is described
+     * by a 3-2-1 intrinsic Tait-Bryan rotation sequence.
+     *
+     * @param euler euler angle instance
+     */
+    Quaternion(const Euler<Type> & euler) :
+        Vector<Type, 4>()
+    {
+        set_from_euler(euler);
+    }
+
+    /**
+     * Constructor from quaternion values
```

Comment:
The quaternion order is missing here - a good way to express this is to provide the zero-rotation / identify example, which is (1, 0, 0, 0).